addons/web/static/src/js/views/calendar/calendar_renderer.js

```diff
@@ -167,6 +168,208 @@ var SidebarFilter = Widget.extend(FieldManagerMixin, {
+         * this view cannot manage them.
+         */
+        unselect() {
+            for (const { calendar } of this.months) {
+                calendar.unselect();
+            }
+        }
+        /**
+         * @override
+         */
+        render() {
+            this.renderSubCalendarsMem(this.context);
+            super.render(...arguments);
+        }
+        /**
+         * Renders the main layout (the 4x3 month grid)
+         */
+        renderSubCalendars() {
+            this.el.classList.add('fc-scroller');
+            this.months = [];
+            for (let monthNumber = 0; monthNumber < 12; monthNumber++) {
+                const monthDate = new Date(this.currentDate.getFullYear(), monthNumber);
+                const monthShortName = moment(monthDate).format('MMM').toLowerCase();
+                const container = createElement('div', { class: 'fc-month-container' });
+                this.el.appendChild(container);
+                const el = createElement('div', {
+                    class: `fc-month fc-month-${monthShortName}`,
+                });
+                container.appendChild(el);
+                const calendar = this._createMonthCalendar(el, monthDate);
+                this.months.push({ el, calendar });
+                calendar.render();
+            }
+        }
+        unrenderSubCalendars() {
+            for (const { el, calendar } of this.months) {
+                calendar.destroy();
+                el.remove();
+            }
+        }
+        renderEvents() {
+            // `renderDates` also renders events so if it's called just before
+            // then do not execute this as it will do a re-render.
+            if (this.datesRendered) {
+                this.datesRendered = false;
+                return;
+            }
+            this.events = this._computeEvents();
+            for (const { calendar } of this.months) {
+                calendar.refetchEvents();
+            }
+        }
+        renderDates() {
+            this.events = this._computeEvents();
+            for (const [monthNumber, { calendar }] of Object.entries(this.months)) {
+                const monthDate = new Date(this.currentDate.getFullYear(), monthNumber);
+                calendar.gotoDate(monthDate);
+            }
+            this.datesRendered = true;
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         */
+        _computeEvents() {
+            const calendar = this.context.calendar;
+            return calendar.getEvents().map(event => {
+                const endUTC = calendar.dateEnv.toDate(event._instance.range.end);
+                const end = new Date(event._instance.range.end);
+                if (endUTC.getHours() > 0 || endUTC.getMinutes() > 0 ||
```

Comment:
not sure it is correct to check this on the UTC date. What if the datetime UTC is 00:00, but in your timezone (UTC+1), it is 01:00? If I understand correctly, you increment the date so that the event is correctly displayed on the day it occurs (even for a single hour). In this case, I don't think it works

Reply:
Date and time were the most annoying thing in this task, especially in this function.
it seems that `endUTC` is the correct datetime
when we check the value of `event.extendedProps.r_end` (which is the date we use for popups)
it's the same as `endUTC` and `end` is `endUTC` + 1h if timezone is UTC+1 so I think `end` is not the correct date

I checked in multiple apps if the dates are correctly displayed and this gives the same display results as the month view